### ISSUE TITLE
Standardized guide-website with reputation system

### ISSUE THREAD

**Issue by AronSchoffer** [open-source-ideas/open-source-ideas#192]
## Project description

A [*Wikipedia*](https://www.wikipedia.org/)-like website where users can read and submit guides for any subject. Using a [*Stack Exchange*](https://stackexchange.com/tour)-like reputation system, the best guides rise to the top and can be edited by anyone with enough reputation. Ensuring higher quality, up-to-date guides. 

### Reputation system
If you don't know what a reputation system is, [this](https://stackexchange.com/tour) explains it quickly and clearly. Of course, it would have to be adjusted to make it suitable for guides.

### Templates
The guides could also be standardized by the use of templates. 
For example, the subject *Python* (the programming language) could use templates like: 
- *Hello World* -- Guide to set you up quickly
- *Classic Guide* -- Full guide teaching you all the basics
- *Landscape* -- Gives perspective to where *Python* falls in the world of programming

### Non-profit
I intend to make the project a non-profit. Some money will need to be made though and I hope to do that without ads. Like [*Wikipedia*](https://www.quora.com/How-does-Wikipedia-earn-money), the project could be funded by donations. I imagine a health bar of sorts on the site which will let users know if there is a need for money.

### The benefits
- Trustworthy guides
Since anyone with enough reputation can edit a guide, guides will contain fewer errors or out-of-date statements. The reputation system will also ensure unbiased guides.
- Standardized guides
All guides will have the same clean design. No ads flashing around. 
Users would be used to the site so more complicated features could be implemented.
Ease of use can also be greatly improved by the use of shortcuts. 
- Meta discussion
There will be a need for discussion about the guides in general. This will naturally push  *guide-knowledge* forward resulting in better guides everywhere!

### Where I need your help
I intend to work on the project myself ([repository](https://github.com/community-guides/community-guides) is in the works) but I'm pretty new at open source and have no experience setting up a big website. Any feedback would be greatly appreciated. Let me know if you want to help!

## Relevant Technology

The project can be seen as a mix of [*Stack Exchange*](https://stackexchange.com), [*Wikipedia*](https://www.wikipedia.org/) and [Tutorialspoint](https://www.tutorialspoint.com/index.htm). 
Because of the scope of this project, I think it would be wise to focus on discussion first, before actual implementation.

### Complexity

- [x] Beginner - This project requires no or little prior knowledge of the technolog(y|ies) specified to contribute to the project
- [ ] Intermediate - The user should have some prior knowledge of the technolog(y|ies) to the point where they know how to use it, but not necessarily all the nooks and crannies of the technology
- [ ] Advanced - The project requires the user to have a good understanding of all components of the project to contribute

### Required time (ETA)

- [ ] Little work - A couple of days
- [ ] Medium work - A week or two
- [x]  Much work - The project will take more than a couple of weeks and serious planning is required

### Categories
I wasn't sure which to choose here.
- [ ] Mobile app
- [ ] IoT
- [ ] Web app
- [ ] Frontend/UI
- [ ] AI/ML
- [ ] APIs/Backend
- [ ] Voice Assistant
- [ ] Developer Tooling
- [ ] Extension/Plugin/Add-On
- [ ] Design/UX
- [ ] AR/VR
- [ ] Bots
- [ ] Security
- [ ] Blockchain
- [ ] Futuristic Tech/Something Unique

**Comment by remram44** [comment]
Sounds like WikiHow and similar? It doesn't seem like there should be multiple such websites (it would just make it harder to find guides) so I'm not sure if there's any benefit in this being open-source 🤔

**Comment by Maikuolan** [comment]
Not sure if it's quite what you're going for, but on the surface, it sounds very similar to [MediaWiki](https://www.mediawiki.org/wiki/MediaWiki).

**Comment by remram44** [comment]
MediaWiki is a generic wiki system, it doesn't have author reputation. However it might be enough for this and in fact already exists in the form of [wikibooks.org](https://www.wikibooks.org/)

**Comment by AronSchoffer** [comment]
Thanks for the feedback, appreciate it! 
I'll take a bit to think about it and look into Wikibooks before I'll respond though.

**Comment by Manishfoodtechs** [comment]
I like your idea of mix. I may agree to fund your project if it is going to
be like 'knowledge base' kind of depository. Further this knowledge base
should get added to any website by API as their knowledge base.

What I mean is described in below example.
Suppose someone is starting a website hosting business then he can get an
API from the project to Integrate free knowledge base. Just check knowledge
base of few hosting websites or a2hosting.

Education institution can also get api for knowledge base for different
subjects etc....

If you and few of our friends get agree we can work on it :). I am in.

**Comment by AronSchoffer** [comment]
@remram44 and @Maikuolan:
It seems Wikibooks is indeed what I was going for. I'll look into it further but for now, we should probably close this issue right? (again, I'm pretty new at all open source stuff)

**Comment by AronSchoffer** [comment]
@Manishfoodtechs I don't really understand what you're saying but it seems like it would be a whole other project. Maybe you should open a new issue with your idea so we close this one.

**Comment by Maikuolan** [comment]
> we should probably close this issue right?

I've been in open source for a while, I guess (relatively speaking), but I'm very new to any kind of participation in this repository (open source ideas), so I can't say much on that note yet. I'm sure someone else will pipe in soon enough though. :-)

**Comment by Maikuolan** [comment]
> Maybe you should open a new issue with your idea so we close this one.

I agree about this. Idea is definitely interesting, and I can see potential for it, but it's not really quite the same as the original idea brought up by this issue.
